sympy/solvers/tests/test_solvers.py

```diff
@@ -2160,9 +2163,25 @@ def test_issue_10993():
     assert solve(x-binomial(a, 3) + binomial(y, 2) + sin(a), x) == [-sin(a) + binomial(a, 3) - binomial(y, 2)]
     assert solve((x+1)-binomial(x+1, 3), x) == [-2, -1, 3]
 
+
 def test_issue_11553():
     eq1 = x + y + 1
     eq2 = x + GoldenRatio
     assert solve([eq1, eq2], x, y) == {x: -GoldenRatio, y: -1 + GoldenRatio}
     eq3 = x + 2 + TribonacciConstant
     assert solve([eq1, eq3], x, y) == {x: -2 - TribonacciConstant, y: 1 + TribonacciConstant}
+
+
+def test_issue_19113_19102():
+    t = S(1)/3
+    solve(cos(x)**5-sin(x)**5)
+    assert solve(4*cos(x)**3 - 2*sin(x)**3) == [
+        atan(2**(t)), -atan(2**(t)*(1 - sqrt(3)*I)/2),
+        -atan(2**(t)*(1 + sqrt(3)*I)/2)]
+    h = S.Half
+    assert solve(cos(x)**2 + sin(x)) == [
+        2*atan(-h + sqrt(5)/2 + sqrt(2)*sqrt(1 - sqrt(5))/2),
+        -2*atan(h + sqrt(5)/2 + sqrt(2)*sqrt(1 + sqrt(5))/2),
+        -2*atan(-sqrt(5)/2 + h + sqrt(2)*sqrt(1 - sqrt(5))/2),
```

Comment:
Here we get two negative real roots. There are 4 real roots in `(0, pi)`. Is that because the equation has been transformed somehow?

Reply:
Yes, the equation becomes `tan(x)/cos(x) + 1` -> `tan(x) + cos(x)`. Let's see if [this link](https://www.desmos.com/calculator/mnyqby8ctk) works. There are two real roots and two imaginary ones.
```
[1.6 + 1.1*I, -2.5, 1.6 - 1.1*I, -0.67]
```